src/Orleans/Streams/Predicates/IStreamNamespacePredicate.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Orleans.Streams
+{
+    public interface IStreamNamespacePredicate
+    {
+        bool IsMatch(string streamNamespace);
```

Comment:
If we're going to support custom filtering, I suggest that we widen the filter to include stream full stream id, that is stream guid, namespace, and provider name.  Something like:

       bool IsMatch(string streamProviderName, IStreamIdentity streamIdentity);